File: Frontend/DiveApp/Models/AuthModels.swift
import Foundation

struct SocialLoginRequest: Encodable {
    let token: String
}

struct TokenResponse: Decodable {
    let accessToken: String
    let refreshToken: String
    let tokenType: String
    let isNewUser: Bool
}

struct RefreshTokenRequest: Encodable {
    let refreshToken: String
}

struct AccessTokenResponse: Decodable {
    let accessToken: String
    let tokenType: String
}

File: Android/app/src/main/java/com/diveapp/android/model/AuthModels.kt
package com.diveapp.android.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class SocialLoginRequest(
    val token: String,
)

@Serializable
data class TokenResponse(
    @SerialName("access_token") val accessToken: String,
    @SerialName("refresh_token") val refreshToken: String,
    @SerialName("token_type") val tokenType: String,
    @SerialName("is_new_user") val isNewUser: Boolean,
)

@Serializable
data class RefreshTokenRequest(
    @SerialName("refresh_token") val refreshToken: String,
)

@Serializable
data class AccessTokenResponse(
    @SerialName("access_token") val accessToken: String,
    @SerialName("token_type") val tokenType: String,
)
